controllers/hostedcontrolplane/hostedcontrolplane.go

```diff
@@ -323,3 +458,277 @@ func (r *HostedControlPlaneReconciler) SetupWithManager(mgr ctrl.Manager) error
+	hostedcontrolplane.SetLabels(labels)
+
+	err := r.Client.Update(ctx, hostedcontrolplane)
+	if err != nil {
+		return fmt.Errorf("error updating hostedcontrolplane monitor: %v", err)
+	}
+	return nil
+}
+
+func (r *HostedControlPlaneReconciler) GetAPIServerHostname(hostedcontrolplane *hypershiftv1beta1.HostedControlPlane, log logr.Logger) (string, error) {
+	for _, service := range hostedcontrolplane.Spec.Services {
+		if service.Service == "APIServer" {
+			return service.ServicePublishingStrategy.Route.Hostname, nil
+		}
+	}
+	return "", fmt.Errorf("APIServer service not found in the hostedcontrolplane")
+}
+
+func (APIClient *APIClient) getDynatraceEquivalentClusterRegionId(hostedcontrolplane *hypershiftv1beta1.HostedControlPlane) (string, error) {
+	// Adapted from spreadsheet in https://issues.redhat.com//browse/SDE-3754
+	// Coming soon regions - il-central-1, ca-west-1
+	awsRegionToDyntraceLocationMapping := map[string]string{
+		"us-east-1":      "N. Virginia",
+		"us-east-2":      "N. Virginia",
+		"us-west-1":      "Oregon",
+		"us-west-2":      "Oregon",
+		"af-south-1":     "São Paulo",
+		"ap-southeast-1": "Singapore",
+		"ap-southeast-2": "Sydney",
+		"ap-southeast-3": "Singapore",
+		"ap-southeast-4": "Sydney",
+		"ap-northeast-1": "Singapore",
+		"ap-northeast-2": "Sydney",
+		"ap-northeast-3": "Singapore",
+		"ap-south-1":     "Mumbai",
+		"ap-south-2":     "Mumbai",
+		"ap-east-1":      "Singapore",
+		"ca-central-1":   "Montreal",
+		"eu-west-1":      "Dublin",
+		"eu-west-2":      "London",
+		"eu-west-3":      "Frankfurt",
+		"eu-central-1":   "Frankfurt",
+		"eu-central-2":   "Frankfurt",
+		"eu-south-1":     "Frankfurt",
+		"eu-south-2":     "Frankfurt",
+		"eu-north-1":     "London",
+		"me-south-1":     "Mumbai",
+		"me-central-1":   "Mumbai",
+		"sa-east-1":      "São Paulo",
+	}
+	clusterRegion := hostedcontrolplane.Spec.Platform.AWS.Region
+	// Look up the dynatrace location name based on the aws region in map
+	locationName, ok := awsRegionToDyntraceLocationMapping[clusterRegion]
+	if !ok {
+		return "", fmt.Errorf("location not found for region: %s", clusterRegion)
+	}
+
+	resp, err := APIClient.makeRequest("GET", "/synthetic/locations", "")
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("failed to fetch locations. Status code: %d", resp.StatusCode)
+	}
+
+	//return location id from response body
+	var locationResponse DynatraceLocation
+	err = json.NewDecoder(resp.Body).Decode(&locationResponse)
+	if err != nil {
+		return "", err
+	}
+	for _, loc := range locationResponse.Locations {
+		if loc.Name == locationName && loc.Type == "PUBLIC" && loc.CloudPlatform == "AMAZON_EC2" {
+			return loc.EntityID, nil
+		}
+	}
+
+	return "", fmt.Errorf("location '%s' not found", locationName)
+}
+
+func (APIClient *APIClient) createDynatraceHTTPMonitor(monitorName, apiUrl, clusterId, dynatraceEquivalentClusterRegionId string) (string, error) {
```

Comment:
If we're renaming `APIClient` to be something like `DynatraceAPIClient` then it might make sense to name some of these functions that are on the struct things like `createHTTPMonitor`.